components/core/src/clp/CurlGlobalInstance.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef CLP_CURLGLOBALINSTANCE_HPP
+#define CLP_CURLGLOBALINSTANCE_HPP
+
+#include <cstddef>
+#include <mutex>
+
+namespace clp {
+/**
+ * Class to wrap `libcurl`'s global initialization/de-initialization calls using RAII. Before using
+ * any `libcurl` functionalities, an instance of this function must be created to ensure underlying
+ * `libcurl` resources have been initialized. This class maintains a static reference count to all
+ * the living instances. De-initialization happens when the reference count reaches 0.
```

Comment:
```suggestion
 * any `libcurl` functionalities, an instance of this class must be created. Although unnecessasry,
 * it can be safely instantiated multiple times; it maintains a static reference count to all
 * existing instances and only di-initializes `libcurl`'s global resources when the reference count
 * reaches 0.
```